src/controls.h

```diff
@@ -2,11 +2,59 @@
 #define CONTROLS_H
 
 #include <sp2/io/keybinding.h>
+#include <cstdio>
+
+class KeyState
+{
+public:
+    KeyState(bool pressed, bool down, bool up, float value) : pressed(pressed), down(down), up(up), value(value) {}
+    KeyState() : pressed(false), down(false), up(false), value(0) {}
+
+    bool pressed;       //True when this key is currently being pressed.
+    bool down;          //True for 1 update cycle when the key is pressed.
+    bool up;            //True for 1 update cycle when the key is released.
+    float value;        //Returns a value in the range -1 to 1 for this keybinding. On keyboard keys this is always 0 or 1, but for joysticks this can be anywhere in the range -1.0 to 1.0
+
+    void writeToFile(FILE* f);
+
+    static bool readFromFile(FILE* f, KeyState& result);
```

Comment:
Why is this a static function?

Reply:
It doesn't need any information from an instance. It could be a non class member function but I find this a good place as well.